### DISCUSSION TITLE
lean-asp-net-core-2-1-manually-setup-a-razor-pages-project-with-bootstrap-npm-and-webpack/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#55] [Blog comments]
# lean-asp-net-core-2-1-manually-setup-a-razor-pages-project-with-bootstrap-npm-and-webpack/

https://blogs.taiga.nl/martijn/posts/lean-asp-net-core-2-1-manually-setup-a-razor-pages-project-with-bootstrap-npm-and-webpack/

**Comment by disqus-comments-migrator** [comment]
<em>[Dan True](https://disqus.com/by/dan_true/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 25 2018, 10:03</em>

---

<p>Thanks for the post. I've been suffering from js/frontend tooling fatigue lately and this was a great primer getting back in the fray.</p><p>One issue though: When I try to use the WebPackDevMiddleware I get the following error after calling "dotnet run watch":</p><p>[1] Ensure that Node.js is installed and can be found in one of the PATH directories.<br>    Current PATH enviroment variable is: <br>    &lt;some of="" my="" path=""&gt; <br>    ;C:\Program Files\nodejs\;C:\Users\&lt;username&gt;\AppData\Roaming\npm;C:\Users\&lt;username&gt;\.dotnet\tools<br>    &lt;some more="" of="" my="" path=""&gt;<br>    Make sure the Node executable is in one of those directories, or update your PATH.</p><p>[2] See the InnerException for further details of the cause.) ---&gt; System.InvalidOperationException: Failed to start Node process. To resolve this:</p><p>If I outcomment the setup code for WebPackDevMiddleware everything runs fine. <br>I can also verify that node is in my path - I can access both npm and node CLI in the directory.</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 25 2018, 10:22, in reply to [Dan True](https://disqus.com/by/dan_true/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953022)</em>

---

<p>Glad you liked the post, thanks. The issue with not finding node.js is usually something environmental. What OS are you using? Which editor? This GitHub issue (<a href="https://github.com/aspnet/JavaScriptServices/issues/707)" rel="nofollow noopener" title="https://github.com/aspnet/JavaScriptServices/issues/707)">https://github.com/aspnet/JavaScriptServices/issues/707)</a> has some potential solutions.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Dan True](https://disqus.com/by/dan_true/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 25 2018, 12:52, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953024)</em>

---

<p>I'm on Windows 10, using Visual Studio 2017. But the above error occurs in my basic command prompt, not through the VS console. But I guess the call to Directory.GetCurrentDirectory() can be the culprit. I will take a look on the github issue.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Dan True](https://disqus.com/by/dan_true/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 25 2018, 01:45, in reply to [Dan True](https://disqus.com/by/dan_true/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953025)</em>

---

<p>Solved. VS intellisense had auto-completed to ConfigurationPath.Combe() instead of Path.Combine(). Silly me.</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 25 2018, 01:49, in reply to [Dan True](https://disqus.com/by/dan_true/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953026)</em>

---

<p>Glad it's solved :)</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Dan True](https://disqus.com/by/dan_true/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 26 2018, 09:17</em>

---

<p>Oh yeah, if you're looking for inspiration for a follow-up you could show how to get typescript working with the above setup (more precisely, with webpack). I'm going through the steps now, but I think a lot of people who are coming here will also be wanting to get typescript into the mix.</p><p>Thanks again!</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 26 2018, 09:38, in reply to [Dan True](https://disqus.com/by/dan_true/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953028)</em>

---

<p>Actually, the next post already involves TypeScript (and React) :-) <a href="https://blogs.taiga.nl/martijn/2018/06/22/lean-asp-net-core-2-1-add-a-react-application-to-an-existing-razor-pages-application/" rel="nofollow noopener" title="https://blogs.taiga.nl/martijn/2018/06/22/lean-asp-net-core-2-1-add-a-react-application-to-an-existing-razor-pages-application/">https://blogs.taiga.nl/martijn/2018/06/22/lean-asp-net-core-2-1-add-a-react-application-to-an-existing-razor-pages-application/</a></p>

**Reply by disqus-comments-migrator** [reply]
<em>[Dan True](https://disqus.com/by/dan_true/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at juni 26 2018, 09:58, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953029)</em>

---

<p>Cool :) I skipped the next one since my frontend-fatigue haven't subsided enough for me to give React a go yet ;) I was just searching for typescript.</p><p>But I got it up and working myself with the ts-loader. Thanks for the nice blog!</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Rory](https://disqus.com/by/disqus_38ZbIR6Ido/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at augustus 01 2018, 03:50</em>

---

<p>launchsettings.json file isn't created by default when you do `dotnet new webapi`. You need to use `dotnet new webapi --use-launch-settings`</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at augustus 01 2018, 04:09, in reply to [Rory](https://disqus.com/by/disqus_38ZbIR6Ido/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953031)</em>

---

<p>Which version of the dotnet sdk are you using (dotnet --version)? Both 2.1.300 and 2.1.302 do generate the launchSettings.json file and have a switch to exclude the launchSettings instead of including.</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Rory](https://disqus.com/by/disqus_38ZbIR6Ido/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at augustus 02 2018, 07:38, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953033)</em>

---

<p>Ah, my mistake I was using version 2.1.202 which confusingly means 2.0 not 2.1. Installing 2.1.302 makes it behave as you describe. Sorry for the distraction.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Preston Hopkins](https://disqus.com/by/prestonhopkins/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 21 2018, 10:53</em>

---

<p>There is one minor typo in this code here:</p><p>npm install -save-dev aspnet-webpack webpack-dev-middleware webpack-hot-middleware</p><p>It should be:</p><p>npm install --save-dev aspnet-webpack webpack-dev-middleware webpack-hot-middleware</p><p>Thanks for this, it was very informative!</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 22 2018, 11:09, in reply to [Preston Hopkins](https://disqus.com/by/prestonhopkins/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953036)</em>

---

<p>Thanks for mentioning. Fixed!</p>

**Reply by disqus-comments-migrator** [reply]
<em>[Preston Hopkins](https://disqus.com/by/prestonhopkins/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at september 23 2018, 08:17, in reply to [Martijn Boland](https://disqus.com/by/martijnboland/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953037)</em>

---

<p>No problem. Your article got me motivated to converting a project of mine to core 2.1 from mvc 5 and to show my team and to move forward with the new iteration of .net. :)</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Adrian Blackburn](https://disqus.com/by/adrian_blackburn/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 17 2018, 11:30</em>

---

<p>Good write-up, thanks. It's probably worth mentioning that you need to include the styles.css file for the production and staging environments in _Layout.cshtml. It's in your screen cap of the file but it think it could be worth clarifying.</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at november 20 2018, 12:07, in reply to [Adrian Blackburn](https://disqus.com/by/adrian_blackburn/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953039)</em>

---

<p>Added to the post. Thanks for mentioning! I missed that one.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[WPlumot](https://disqus.com/by/wplumot/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at december 16 2018, 09:28</em>

---

<p>Great, I was struggling for several hours not knowing what was the best solution. Super.</p>

**Comment by disqus-comments-migrator** [comment]
<em>[Vedran Mandić](https://disqus.com/by/vmandic/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 14 2019, 09:43</em>

---

<p>truly invaluable post, just the right stuff I need for the hybrid approach. I love the Razor stuff and utilize it to the max, but when a complex UI pops out I shift to TS/React/Redux and develop a router of its own for the section of the app, thanks for the excellent overview and finally a walkthrough without using the default template which hides away too much with CRA</p>

**Comment by disqus-comments-migrator** [comment]
<em>[mattiassantoro](https://disqus.com/by/mattiassantoro/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at april 13 2019, 09:30</em>

---

<p>Hi all! Thanks for this tutorial, it was really helpful!!<br>I have just two questions:<br>1) I don't get why in the BuildClientAssets target the include path is &lt;distfiles include="$(ClientAppFolder)\wwwroot\dist\**"/&gt;. If so, you tell to msbuild that the "wwwroot" folder is inside the ClientAppFolder but is on the parent level. (Infact, in webpack.config.js you build in  ..\wwwroot\dist\.<br><s>2) I have some problem with Release configuration on Visual Studio Code. When debugging in release with VS2017 it seems that before running the Webapp it launches the BuildClientAssets target and so the content of wwwroot folder is updated with the new bundle. But if I use VSCode and from terminal I launch dotnet run -c Release the wwwroot folder is not updated, I have to previously launch npm run prod before. <br>Some hint?</s> EDIT: Not true, I was running in Release but with environment set to development so it was the Webpack middleware responsible of the asset build.</p>

**Reply by martijnboland** [reply]
<em>[Martijn Boland](https://disqus.com/by/martijnboland/) commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at april 14 2019, 10:27, in reply to [mattiassantoro](https://disqus.com/by/mattiassantoro/)'s [comment](https://github.com/martijnboland/blog/discussions/55#discussioncomment-8953046)</em>

---

<p>1) You spotted a bug in the .csproj file. The $(ClientAppFolder)\ prefix shouldn't be there. Changed it to &lt;distfiles include="wwwroot\dist\**"/&gt;. Thanks!</p>
